src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -944,6 +924,42 @@ private SortVersion GetSortVersion()
                                                              (byte) (LCID  & 0xFF)));
         }
 
+        private static class SortHandleCache
+        {
+            // in most scenarios there is a limited number of cultures with limited number of sort options
+            // so caching the sort handles and not freeing them is OK, see https://github.com/dotnet/coreclr/pull/25117 for more
+            private static readonly Dictionary<string, IntPtr> s_sortNameToSortHandleCache = new Dictionary<string, IntPtr>();
+
+            internal static IntPtr GetCachedSortHandle(string sortName)
+            {
+                lock (s_sortNameToSortHandleCache)
+                {
+                    if (!s_sortNameToSortHandleCache.TryGetValue(sortName, out IntPtr result))
+                    {
+                        var resultCode = Interop.Globalization.GetSortHandle(sortName, out result);
```

Comment:
```suggestion
                        ResultCode resultCode = Interop.Globalization.GetSortHandle(sortName, out result);
```

The CoreFX coding convention allow 'var' when the type is obvious from the right side.

Reply:
But in this particular case we have to use full type name:

```cs
Interop.Globalization.ResultCode resultCode = Interop.Globalization.GetSortHandle(sortName, out result);
```

imho a `var` combined with good variable name is better:

```cs
var resultCode = Interop.Globalization.GetSortHandle(sortName, out result);
```

Reply:
Please don't use var. This name tells me nothing about what kind of result code it is, for example.

Reply:
I removed the `var`